## 12% of end-end tests have passed
```
Total executed tests: 48
Failed tests: 41
Expected to fail tests: 1
Passed tests: 6
```
```
IDs of failed tests: 702809,702782,702733,702784,703194,702859,702777,702732,702813,703133,702957,702851,702840,703495,703086,702948,702894,702745,702783,702850,703297,704615,703496,702936,702855,703202,703391,702786,702839,702947,702838,702844,702843,702807,702731,702808,702730,702845,702958,703629,702841 
```
```
IDs of expected to fail tests: 703503 
```
<h3>Failed tests (41)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=18901&TR_CASE_IDS=702809,702782,702733,702784,703194,702859,702777,702732,702813,703133,702957,702851,702840,703495,703086,702948,702894,702745,702783,702850,703297,704615,703496,702936,702855,703202,703391,702786,702839,702947,702838,702844,702843,702807,702731,702808,702730,702845,702958,703629,702841">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469247">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469242">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469254">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469244">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469265">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469280">test_1_1_chat_send_image_save_and_share</a>, id: 703391 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469240">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469239">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:40: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9858d25d183447aba3d7af773d050f8d?auth=30f53416c275656d05ac01b0b4c7e6e1">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3154060b2f844f6c9928063e7be832af?auth=619d9bc98fb9ecb045085e133822539a">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469241">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_group_chat.py:55: in prepare_devices
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:44: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user admin')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469281">test_group_chat_mute_chat</a>, id: 703495 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_group_chat.py:55: in prepare_devices
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:44: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user admin')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469278">test_group_chat_send_image_save_and_share</a>, id: 703297 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_group_chat.py:55: in prepare_devices
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:44: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user admin')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469277">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_group_chat.py:55: in prepare_devices
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:44: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user admin')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469251">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><p><blockquote><div>Device 3: Tap on found: Button </div><div>Device 3: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
Test setup failed: critical/chats/test_group_chat.py:55: in prepare_devices
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:44: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user admin')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/35f1c8c47d2d41ba8c88b5c02437ca49?auth=26471e2fae3de739415cb4a530dc0ca7">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/161e92640837458dac3497a6875f5fd7?auth=e7a1664edc86806f535592fc9d2995b5">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5b5610c05796433a97b9af34ba4a95e5?auth=c43d6a46ec49917137f76288dbde3a6a">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469252">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_group_chat.py:55: in prepare_devices
    self.loop.run_until_complete(
/usr/lib/python3.10/asyncio/base_events.py:649: in run_until_complete
    return future.result()
__init__.py:44: in run_in_parallel
    returns.append(await k)
/usr/lib/python3.10/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user admin')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMergedTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469253">test_community_markdown_support</a>, id: 702809 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:818: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469271">test_community_hashtag_links_to_community_channels</a>, id: 702948 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:818: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469250">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:818: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a09033ae29bc4dd78b4e20f4a30e0b3e?auth=acc8a2de1ddc2aea7756f894b85dce3e">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d001528816c24e7bbac7f340ad4d0a81?auth=241f8acb2cd663e8a4aedc7092ac7b38">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469261">test_community_leave</a>, id: 702845 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:818: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469284">test_community_join_when_node_owner_offline</a>, id: 703629 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:818: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469246">test_add_contact_field_validation</a>, id: 702777 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
activity_center/test_activity_center.py:190: in test_add_contact_field_validation
    self.home_2.handle_contact_request(new_username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'test user 123')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/15414973220e45c3ad69848921cee8d9?auth=0689953c7db0c8b7ea7b1c9faf96c2a6#87">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/373a5e205a944172b9ce92e4cfe7c625?auth=1292e421cfb27eeb1340c23761bed293#154">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469264">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `ChatsTab` by `accessibility id`: `chats-stack-tab` </div><div>Device 1: Tap on found: ChatsTab </div></blockquote></p>

```
activity_center/test_activity_center.py:112: in test_activity_center_contact_request_accept_swipe_mark_all_as_read
    self.home_1.notifications_unread_badge.wait_for_visibility_of_element(30)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 1: BaseElement by accessibility id:`activity-center-unread-count` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/15414973220e45c3ad69848921cee8d9?auth=0689953c7db0c8b7ea7b1c9faf96c2a6#75">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/373a5e205a944172b9ce92e4cfe7c625?auth=1292e421cfb27eeb1340c23761bed293#87">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469263">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div>Device 1: Find `Text` by `xpath`: `//*[@content-desc="pending-contact-requests-count"]/android.widget.TextView` </div></blockquote></p>

```
activity_center/test_activity_center.py:69: in test_activity_center_contact_request_decline
    if self.home_1.pending_contact_request_text.text != '1':
../views/base_element.py:415: in text
    text = self.find_element().text
../views/base_element.py:79: in find_element
    raise NoSuchElementException(
 Device 1: Text by xpath: `//*[@content-desc="pending-contact-requests-count"]/android.widget.TextView` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/15414973220e45c3ad69848921cee8d9?auth=0689953c7db0c8b7ea7b1c9faf96c2a6#59">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/373a5e205a944172b9ce92e4cfe7c625?auth=1292e421cfb27eeb1340c23761bed293#58">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePRTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469272">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
Test setup failed: activity_center/test_activity_center.py:393: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a918a79d4135489b8374f6bab85bc75b?auth=bdb87e331aab811e848c58a9ed2018b2">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f523bfd1af6f47f09931049380bac440?auth=47f8fd25a7814d1c2a39be795dc5068c">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469273">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td>

```
Test setup failed: activity_center/test_activity_center.py:393: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUiTwo:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469249">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:546: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469248">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:546: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8b3e21eb20a3461cb95dd3be575f68ef?auth=d2ef3262dbef4d38d52ed660dcb97c7e">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5502de3cfc264d3ea6c2071f2b82977b?auth=ca4cb6245b77a69b52c20e67b8ee6c6f">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469282">test_1_1_chat_mute_chat</a>, id: 703496 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:546: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'sender')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469276">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469266">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469257">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469274">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469268">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469286">test_community_edit_delete_message_when_offline</a>, id: 704615 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469256">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469255">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2ffc9926b34444a28ed5e8d71116974f?auth=3c3f163af6d231571f15293e27ca5264">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7a45356f71f54ce58b1adf207e4fda7b?auth=28545d942c6ed8dc7200e806cdc0dade">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>9. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469260">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>10. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469259">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr><tr><td><b>11. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469258">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_public_chat_browsing.py:296: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user_1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469269">test_navigation_jump_to</a>, id: 702936 </b></td></tr><tr><td><p><blockquote><div>Device 2: Tap on found: Button </div><div>Device 2: Attempt 0 is successful clicking close-activity-center </div></blockquote></p>

```
Test setup failed: activity_center/test_activity_center.py:246: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/4be6a72afe024a2894f363d62894facf?auth=7fcaf5be174097aba4276954b63b1b58">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/f296746e6448487282cc707cf5f6da11?auth=e90254027d4be54b929a8ecdbadf50fd">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469270">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td>

```
Test setup failed: activity_center/test_activity_center.py:246: in prepare_devices
    self.home_2.handle_contact_request(self.username_1)
../views/home_view.py:382: in handle_contact_request
    chat_element.accept_contact_request()
../views/home_view.py:150: in accept_contact_request
    self.handle_cr("accept-contact-request")
../views/home_view.py:147: in handle_cr
    ).wait_for_rendering_ended_and_click()
../views/base_element.py:163: in wait_for_rendering_ended_and_click
    self.wait_for_visibility_of_element(20)
../views/base_element.py:147: in wait_for_visibility_of_element
    raise TimeoutException(
 Device 2: Button by xpath:`//*[contains(@text, 'user1')]/ancestor::*[@content-desc='activity']/*[@content-desc="accept-contact-request"]` is not found on the screen after wait_for_visibility_of_element
```

<br/><br/></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469275">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `show-profiles` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:276: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.errors.verify_no_errors()
base_test_case.py:191: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 zQ3...dWXh5 was not restored as a contact from waku backup!
E    zQ3...Vacac was not restored as a contact from waku backup!
E    admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c78da52de1804f5b80a2c1e6a8a75275?auth=7f7ce3d6b590cdb62da4c7c734993434#252">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469283">test_community_discovery</a>, id: 703503 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Curated communities not loading, https://github.com/status-im/status-mobile/issues/17852]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (6)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469243">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c78da52de1804f5b80a2c1e6a8a75275?auth=7f7ce3d6b590cdb62da4c7c734993434#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469267">test_community_undo_delete_message</a>, id: 702869 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c78da52de1804f5b80a2c1e6a8a75275?auth=7f7ce3d6b590cdb62da4c7c734993434#108">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469262">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c78da52de1804f5b80a2c1e6a8a75275?auth=7f7ce3d6b590cdb62da4c7c734993434#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469279">test_community_mute_community_and_channel</a>, id: 703382 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/c78da52de1804f5b80a2c1e6a8a75275?auth=7f7ce3d6b590cdb62da4c7c734993434#127">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeepLinksOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469285">test_links_open_universal_links_from_chat</a>, id: 704613 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/704ee0527b56496190b5cfcb56a6ae04?auth=daf03cc77143ad191dd87b36b6873885#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3469245">test_links_deep_links</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/704ee0527b56496190b5cfcb56a6ae04?auth=daf03cc77143ad191dd87b36b6873885#199">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/quo/components/browser/browser_input/view.cljs
+++ b/src/quo/components/browser/browser_input/view.cljs
@@ -6,8 +6,7 @@
     [quo.foundations.colors :as colors]
     [quo.theme :as quo.theme]
     [react-native.core :as rn]
-    [react-native.platform :as platform]
-    [reagent.core :as reagent]))
+    [react-native.platform :as platform]))
 
 (defn remove-http-https-www
   [value]
@@ -68,78 +67,86 @@
   [:cursor-color :placeholder-text-color :editable :on-change-text :on-focus
    :on-blur :on-clear :value :disabled? :blur? :customization-color :theme])
 
-(defn- view-internal
-  [{:keys [default-value]
-    :or   {default-value ""}}]
-  (let [state       (reagent/atom :default)
-        value       (reagent/atom default-value)
-        set-active  #(reset! state :active)
-        set-default #(reset! state :default)
-        set-value   #(reset! value %)
-        ref         (atom nil)
-        clear-input (fn []
-                      (.clear ^js @ref)
-                      (reset! value ""))
-        focus-input (fn []
-                      (set-active)
-                      (.focus ^js @ref))]
-    (fn [{:keys [disabled? blur? on-change-text customization-color
-                 on-clear on-focus on-blur theme get-ref locked?
-                 favicon favicon-color favicon-size]
-          :as   props}]
-      (let [clean-props (apply dissoc props props-to-remove)]
-        [rn/view {:style style/root-container}
-         (when (and (seq @value) (= @state :default))
-           [rn/touchable-opacity
-            {:style    style/default-container
-             :on-press focus-input}
-            (when favicon
-              [icon/icon favicon
-               {:accessibility-label :browser-input-favicon
-                :color               favicon-color
-                :container-style     style/favicon-icon-container
-                :size                favicon-size}])
-            [rn/text
-             {:accessibility-label :browser-input-label
-              :style               (style/text)} (remove-http-https-www @value)]
-            (when locked?
-              [lock-icon
-               {:blur? blur?
-                :theme theme}])])
-         [rn/view {:style (style/active-container (or (empty? @value) (= @state :active)))}
-          [rn/text-input
-           (merge
-            clean-props
-            {:accessibility-label    :browser-input
-             :auto-capitalize        :none
-             :auto-correct           false
-             :cursor-color           (cursor-color customization-color theme)
-             :editable               (not disabled?)
-             :keyboard-appearance    (colors/theme-colors :light :dark theme)
-             :keyboard-type          :web-search
-             :on-blur                (fn []
-                                       (set-default)
-                                       (when on-blur (on-blur)))
-             :on-change-text         (fn [new-text]
-                                       (set-value new-text)
-                                       (when on-change-text (on-change-text new-text)))
-             :on-focus               (fn []
-                                       (set-active)
-                                       (when on-focus (on-focus)))
-             :placeholder-text-color (placeholder-color @state blur? theme)
-             :ref                    (fn [r]
-                                       (reset! ref r)
-                                       (when get-ref (get-ref r)))
-             :selection-color        (when platform/ios?
-                                       (cursor-color customization-color theme))
-             :select-text-on-focus   true
-             :style                  (style/input disabled?)})]
-          (when (seq @value)
-            [clear-button
-             {:blur?    blur?
-              :on-press (fn []
-                          (clear-input)
-                          (when on-clear (on-clear)))
-              :theme    theme}])]]))))
-
-(def view (quo.theme/with-theme view-internal))
+(defn view
+  [{:keys [disabled? blur? on-change-text customization-color
+           on-clear on-focus on-blur get-ref locked?
+           favicon favicon-color favicon-size default-value]
+    :or   {default-value ""}
+    :as   props}]
+  (let [ref               (rn/use-ref-atom nil)
+        on-ref            (rn/use-callback
+                           (fn [r]
+                             (reset! ref r)
+                             (when get-ref (get-ref r)))
+                           [get-ref])
+        theme             (quo.theme/use-theme)
+        [state set-state] (rn/use-state :default)
+        [value set-value] (rn/use-state default-value)
+        on-clear          (rn/use-callback
+                           (fn []
+                             (.clear ^js @ref)
+                             (set-value "")
+                             (when on-clear (on-clear)))
+                           [on-clear])
+        focus-input       (rn/use-callback
+                           (fn []
+                             (set-state :active)
+                             (.focus ^js @ref)))
+        on-blur           (rn/use-callback
+                           (fn []
+                             (set-state :default)
+                             (when on-blur (on-blur)))
+                           [on-blur])
+        on-change-text    (rn/use-callback
+                           (fn [new-text]
+                             (set-value new-text)
+                             (when on-change-text (on-change-text new-text)))
+                           [on-change-text])
+        on-focus          (rn/use-callback
+                           (fn []
+                             (set-state :active)
+                             (when on-focus (on-focus)))
+                           [on-focus])
+        clean-props       (apply dissoc props props-to-remove)]
+    [rn/view {:style style/root-container}
+     (when (and (seq value) (= state :default))
+       [rn/touchable-opacity
+        {:style    style/default-container
+         :on-press focus-input}
+        (when favicon
+          [icon/icon favicon
+           {:accessibility-label :browser-input-favicon
+            :color               favicon-color
+            :container-style     style/favicon-icon-container
+            :size                favicon-size}])
+        [rn/text
+         {:accessibility-label :browser-input-label
+          :style               (style/text)}
+         (remove-http-https-www value)]
+        (when locked?
+          [lock-icon {:blur? blur? :theme theme}])])
+     [rn/view {:style (style/active-container (or (empty? value) (= state :active)))}
+      [rn/text-input
+       (merge
+        clean-props
+        {:accessibility-label    :browser-input
+         :auto-capitalize        :none
+         :auto-correct           false
+         :cursor-color           (cursor-color customization-color theme)
+         :editable               (not disabled?)
+         :keyboard-appearance    (colors/theme-colors :light :dark theme)
+         :keyboard-type          :web-search
+         :on-blur                on-blur
+         :on-change-text         on-change-text
+         :on-focus               on-focus
+         :placeholder-text-color (placeholder-color state blur? theme)
+         :ref                    on-ref
+         :selection-color        (when platform/ios?
+                                   (cursor-color customization-color theme))
+         :select-text-on-focus   true
+         :style                  (style/input disabled?)})]
+      (when (seq value)
+        [clear-button
+         {:blur?    blur?
+          :on-press on-clear
+          :theme    theme}])]]))

--- a/src/quo/components/buttons/button/view.cljs
+++ b/src/quo/components/buttons/button/view.cljs
@@ -7,10 +7,9 @@
     [quo.foundations.customization-colors :as customization-colors]
     [quo.theme :as theme]
     [react-native.blur :as blur]
-    [react-native.core :as rn]
-    [reagent.core :as reagent]))
+    [react-native.core :as rn]))
 
-(defn- button-internal
+(defn button
   "with label
    [button opts \"label\"]
    opts
@@ -29,102 +28,103 @@
     :theme :light/:dark
    only icon
    [button {:icon-only? true} :i/close-circle]"
-  [_ _]
-  (let [pressed-state? (reagent/atom false)]
-    (fn
-      [{:keys [on-press on-long-press disabled? type background size icon-left icon-right icon-top
-               customization-color theme accessibility-label icon-only? container-style inner-style
-               pressed? on-press-in on-press-out allow-multiple-presses?]
-        :or   {type                :primary
-               size                40
-               customization-color (if (= type :primary) :blue nil)}}
-       children]
-      (let [{:keys [icon-color background-color label-color border-color blur-type
-                    blur-overlay-color border-radius overlay-customization-color]}
-            (button-properties/get-values {:customization-color customization-color
-                                           :background          background
-                                           :type                type
-                                           :theme               theme
-                                           :pressed?            (if pressed? pressed? @pressed-state?)
-                                           :icon-only?          icon-only?})
-            icon-size (when (= 24 size) 12)]
-        [rn/touchable-without-feedback
-         {:disabled                disabled?
-          :accessibility-label     accessibility-label
-          :on-press-in             (fn []
-                                     (reset! pressed-state? true)
-                                     (when on-press-in (on-press-in)))
-          :on-press-out            (fn []
-                                     (reset! pressed-state? nil)
-                                     (when on-press-out (on-press-out)))
-          :on-press                on-press
-          :allow-multiple-presses? allow-multiple-presses?
-          :on-long-press           on-long-press}
+  [{:keys [on-press on-long-press disabled? type background size icon-left icon-right icon-top
+           customization-color accessibility-label icon-only? container-style inner-style
+           pressed? on-press-in on-press-out allow-multiple-presses?]
+    :or   {type                :primary
+           size                40
+           customization-color (if (= type :primary) :blue nil)}}
+   children]
+  (let [[pressed-state? set-pressed-state] (rn/use-state false)
+        theme (theme/use-theme-value)
+        {:keys [icon-color background-color label-color border-color blur-type
+                blur-overlay-color border-radius overlay-customization-color]}
+        (button-properties/get-values {:customization-color customization-color
+                                       :background          background
+                                       :type                type
+                                       :theme               theme
+                                       :pressed?            (if pressed? pressed? pressed-state?)
+                                       :icon-only?          icon-only?})
+        icon-size (when (= 24 size) 12)
+        on-press-in-cb (rn/use-callback
+                        (fn []
+                          (set-pressed-state true)
+                          (when on-press-in (on-press-in))))
+        on-press-out-cb (rn/use-callback
+                         (fn []
+                           (set-pressed-state nil)
+                           (when on-press-out (on-press-out))))]
+    [rn/touchable-without-feedback
+     {:disabled                disabled?
+      :accessibility-label     accessibility-label
+      :on-press-in             on-press-in-cb
+      :on-press-out            on-press-out-cb
+      :on-press                on-press
+      :allow-multiple-presses? allow-multiple-presses?
+      :on-long-press           on-long-press}
+     [rn/view
+      {:style (merge
+               (style/shape-style-container size border-radius)
+               container-style)}
+      [rn/view
+       {:style (merge
+                (style/style-container {:size             size
+                                        :disabled?        disabled?
+                                        :border-radius    border-radius
+                                        :background-color background-color
+                                        :border-color     border-color
+                                        :icon-only?       icon-only?
+                                        :icon-top         icon-top
+                                        :icon-left        icon-left
+                                        :icon-right       icon-right})
+                inner-style)}
+       (when overlay-customization-color
+         [customization-colors/overlay
+          {:customization-color overlay-customization-color
+           :theme               theme
+           :pressed?            (if pressed? pressed? pressed-state?)}])
+       (when (= background :photo)
+         [blur/view
+          {:blur-radius   20
+           :blur-type     blur-type
+           :overlay-color blur-overlay-color
+           :style         style/blur-view}])
+       (when icon-top
          [rn/view
-          {:style (merge
-                   (style/shape-style-container size border-radius)
-                   container-style)}
-          [rn/view
-           {:style (merge
-                    (style/style-container {:size             size
-                                            :disabled?        disabled?
-                                            :border-radius    border-radius
-                                            :background-color background-color
-                                            :border-color     border-color
-                                            :icon-only?       icon-only?
-                                            :icon-top         icon-top
-                                            :icon-left        icon-left
-                                            :icon-right       icon-right})
-                    inner-style)}
-           (when overlay-customization-color
-             [customization-colors/overlay
-              {:customization-color overlay-customization-color
-               :theme               theme
-               :pressed?            (if pressed? pressed? @pressed-state?)}])
-           (when (= background :photo)
-             [blur/view
-              {:blur-radius   20
-               :blur-type     blur-type
-               :overlay-color blur-overlay-color
-               :style         style/blur-view}])
-           (when icon-top
-             [rn/view
-              [quo.icons/icon icon-top
-               {:container-style {:margin-bottom 2}
-                :color           icon-color
-                :size            icon-size}]])
-           (when icon-left
-             [rn/view
-              {:style (style/icon-left-icon-style
-                       {:size      size
-                        :icon-size icon-size})}
-              [quo.icons/icon icon-left
-               {:color icon-color
-                :size  icon-size}]])
-           [rn/view
-            (cond
-              icon-only?
-              [quo.icons/icon children
-               {:color label-color
-                :size  icon-size}]
+          [quo.icons/icon icon-top
+           {:container-style {:margin-bottom 2}
+            :color           icon-color
+            :size            icon-size}]])
+       (when icon-left
+         [rn/view
+          {:style (style/icon-left-icon-style
+                   {:size      size
+                    :icon-size icon-size})}
+          [quo.icons/icon icon-left
+           {:color icon-color
+            :size  icon-size}]])
+       [rn/view
+        (cond
+          icon-only?
+          [quo.icons/icon children
+           {:color label-color
+            :size  icon-size}]
 
-              (string? children)
-              [text/text
-               {:size            (when (#{56 24} size) :paragraph-2)
-                :weight          :medium
-                :number-of-lines 1
-                :style           {:color label-color}}
-               children]
+          (string? children)
+          [text/text
+           {:size            (when (#{56 24} size) :paragraph-2)
+            :weight          :medium
+            :number-of-lines 1
+            :style           {:color label-color}}
+           children]
 
-              (vector? children)
-              children)]
-           (when icon-right
-             [rn/view
-              {:style (style/icon-right-icon-style
-                       {:size      size
-                        :icon-size icon-size})}
-              [quo.icons/icon icon-right
-               {:color icon-color
-                :size  icon-size}]])]]]))))
-
-(def button (theme/with-theme button-internal))
+          (vector? children)
+          children)]
+       (when icon-right
+         [rn/view
+          {:style (style/icon-right-icon-style
+                   {:size      size
+                    :icon-size icon-size})}
+          [quo.icons/icon icon-right
+           {:color icon-color
+            :size  icon-size}]])]]]))

--- a/src/quo/components/buttons/composer_button/view.cljs
+++ b/src/quo/components/buttons/composer_button/view.cljs
@@ -3,28 +3,24 @@
     [quo.components.buttons.composer-button.style :as style]
     [quo.components.icon :as quo.icons]
     [quo.theme :as theme]
-    [react-native.core :as rn]
-    [reagent.core :as reagent]))
+    [react-native.core :as rn]))
 
-(defn- view-internal
-  [_ _]
-  (let [pressed? (reagent/atom false)]
-    (fn
-      [{:keys [on-press on-long-press disabled? theme blur? icon accessibility-label container-style]}]
-      [rn/pressable
-       {:accessibility-label (or accessibility-label :composer-button)
-        :on-press            on-press
-        :on-press-in         #(reset! pressed? true)
-        :on-press-out        #(reset! pressed? nil)
-        :on-long-press       on-long-press
-        :disabled            disabled?
-        :style               (merge (style/main {:pressed?  @pressed?
-                                                 :blur?     blur?
-                                                 :theme     theme
-                                                 :disabled? disabled?})
-                                    container-style)}
-       [quo.icons/icon icon
-        {:color (style/get-label-color {:blur? blur?
-                                        :theme theme})}]])))
-
-(def view (theme/with-theme view-internal))
+(defn view
+  [{:keys [on-press on-long-press disabled? blur? icon accessibility-label container-style]}]
+  (let [[pressed? set-pressed] (rn/use-state false)
+        theme                  (theme/use-theme-value)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:accessibility-label (or accessibility-label :composer-button)
+      :on-press            on-press
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out
+      :on-long-press       on-long-press
+      :disabled            disabled?
+      :style               (merge (style/main {:pressed?  pressed?
+                                               :blur?     blur?
+                                               :theme     theme
+                                               :disabled? disabled?})
+                                  container-style)}
+     [quo.icons/icon icon {:color (style/get-label-color {:blur? blur? :theme theme})}]]))

--- a/src/quo/components/buttons/dynamic_button/view.cljs
+++ b/src/quo/components/buttons/dynamic_button/view.cljs
@@ -5,8 +5,7 @@
     [quo.components.markdown.text :as text]
     [quo.foundations.colors :as colors]
     [quo.theme :as quo.theme]
-    [react-native.core :as rn]
-    [reagent.core :as reagent]))
+    [react-native.core :as rn]))
 
 (defn- get-button-color
   [{:keys [type pressed? customization-color theme]}]
@@ -45,48 +44,48 @@
     :color           (get-icon-and-text-color type theme)
     :container-style (style/container type)}])
 
-(defn- view-internal
+(defn view
   "[dynamic-button opts]
    opts
    {:type                :jump-to/:mention/:notification-down/:notification-up/:search/:search-with-label/:scroll-to-bottom
     :on-press            fn
     :count               mentions or notifications count
     :customization-color customize jump-to and mention button color}"
-  [_]
-  (let [pressed? (reagent/atom false)]
-    (fn [{:keys [type label on-press customization-color style theme] :as args}]
-      [rn/touchable-opacity
-       {:on-press-in         #(reset! pressed? true)
-        :on-press-out        #(reset! pressed? false)
-        :on-press            on-press
-        :active-opacity      1
-        :hit-slop            {:top 5 :bottom 5 :left 5 :right 5}
-        :pointer-events      :auto
-        :style               {:height 24}
-        :accessibility-label type}
-       [rn/view
-        {:style (merge
-                 {:flex-direction   :row
-                  :height           24
-                  :border-radius    12
-                  :background-color (get-button-color {:type                type
-                                                       :pressed?            @pressed?
-                                                       :customization-color (or customization-color
-                                                                                :primary)
-                                                       :theme               theme})}
-                 style)}
-        (when (#{:mention :search :search-with-label :scroll-to-bottom} type)
-          [icon-view type])
-        (when (#{:jump-to :mention :notification-down :notification-up :search-with-label} type)
-          [text/text
-           {:weight :medium
-            :size   :paragraph-2
-            :style  (assoc (style/text type) :color (get-icon-and-text-color type theme))}
-           (case type
-             :jump-to                                       label
-             :search-with-label                             label
-             (:mention :notification-down :notification-up) (str (:count args)))])
-        (when (#{:jump-to :notification-down :notification-up} type)
-          [icon-view type theme])]])))
-
-(def view (quo.theme/with-theme view-internal))
+  [{:keys [type label on-press customization-color style] :as args}]
+  (let [theme                  (quo.theme/use-theme-value)
+        [pressed? set-pressed] (rn/use-state false)
+        button-color           (get-button-color {:type                type
+                                                  :pressed?            pressed?
+                                                  :customization-color (or customization-color :primary)
+                                                  :theme               theme})
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed false))]
+    [rn/touchable-opacity
+     {:on-press-in         on-press-in
+      :on-press-out        on-press-out
+      :on-press            on-press
+      :active-opacity      1
+      :hit-slop            {:top 5 :bottom 5 :left 5 :right 5}
+      :pointer-events      :auto
+      :style               {:height 24}
+      :accessibility-label type}
+     [rn/view
+      {:style (merge
+               {:flex-direction   :row
+                :height           24
+                :border-radius    12
+                :background-color button-color}
+               style)}
+      (when (#{:mention :search :search-with-label :scroll-to-bottom} type)
+        [icon-view type])
+      (when (#{:jump-to :mention :notification-down :notification-up :search-with-label} type)
+        [text/text
+         {:weight :medium
+          :size   :paragraph-2
+          :style  (assoc (style/text type) :color (get-icon-and-text-color type theme))}
+         (case type
+           :jump-to                                       label
+           :search-with-label                             label
+           (:mention :notification-down :notification-up) (str (:count args)))])
+      (when (#{:jump-to :notification-down :notification-up} type)
+        [icon-view type theme])]]))

--- a/src/quo/components/buttons/logout_button/view.cljs
+++ b/src/quo/components/buttons/logout_button/view.cljs
@@ -6,29 +6,25 @@
     [quo.foundations.colors :as colors]
     [quo.theme :as quo.theme]
     [react-native.core :as rn]
-    [reagent.core :as reagent]
     [utils.i18n :as i18n]))
 
-(defn- view-internal
-  [_]
-  (let [pressed?     (reagent/atom false)
-        on-press-in  #(reset! pressed? true)
-        on-press-out #(reset! pressed? nil)]
-    (fn
-      [{:keys [on-press on-long-press disabled? theme container-style]}]
-      [rn/pressable
-       {:accessibility-label :log-out-button
-        :on-press            on-press
-        :on-press-in         on-press-in
-        :on-press-out        on-press-out
-        :on-long-press       on-long-press
-        :disabled            disabled?
-        :style               (merge (style/main {:pressed?  @pressed?
-                                                 :theme     theme
-                                                 :disabled? disabled?})
-                                    container-style)}
-       [icon/icon :i/log-out {:color (if pressed? colors/white-opa-40 colors/white-opa-70)}]
-       [text/text {:weight :medium :size :paragraph-1}
-        (i18n/label :t/logout)]])))
-
-(def view (quo.theme/with-theme view-internal))
+(defn view
+  [{:keys [on-press on-long-press disabled? container-style]}]
+  (let [theme                  (quo.theme/use-theme-value)
+        [pressed? set-pressed] (rn/use-state false)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:accessibility-label :log-out-button
+      :on-press            on-press
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out
+      :on-long-press       on-long-press
+      :disabled            disabled?
+      :style               (merge (style/main {:pressed?  pressed?
+                                               :theme     theme
+                                               :disabled? disabled?})
+                                  container-style)}
+     [icon/icon :i/log-out {:color (if pressed? colors/white-opa-40 colors/white-opa-70)}]
+     [text/text {:weight :medium :size :paragraph-1}
+      (i18n/label :t/logout)]]))

--- a/src/quo/components/buttons/slide_button/animations.cljs
+++ b/src/quo/components/buttons/slide_button/animations.cljs
@@ -1,9 +1,5 @@
 (ns quo.components.buttons.slide-button.animations
-  (:require
-    [oops.core :as oops]
-    [quo.components.buttons.slide-button.utils :as utils]
-    [react-native.gesture :as gesture]
-    [react-native.reanimated :as reanimated]))
+  (:require [react-native.reanimated :as reanimated]))
 
 (def ^:private extrapolation
   {:extrapolateLeft  "clamp"
@@ -66,36 +62,6 @@
                                                 :damping   30
                                                 :stiffness 400}))
 
-(defn- complete-animation
-  [sliding-complete?]
-  (reset! sliding-complete? true))
-
 (defn reset-track-position
   [x-pos]
   (animate-spring x-pos 0))
-
-;; Gestures
-(defn drag-gesture
-  [x-pos
-   gestures-disabled?
-   disabled?
-   track-width
-   sliding-complete?]
-  (let [gestures-enabled? (not (or disabled? @gestures-disabled?))]
-    (-> (gesture/gesture-pan)
-        (gesture/with-test-ID :slide-button-gestures)
-        (gesture/enabled gestures-enabled?)
-        (gesture/min-distance 0)
-        (gesture/on-update (fn [event]
-                             (let [x-translation (oops/oget event "translationX")
-                                   clamped-x     (utils/clamp-value x-translation 0 track-width)
-                                   reached-end?  (>= clamped-x track-width)]
-                               (reanimated/set-shared-value x-pos clamped-x)
-                               (when (and reached-end? (not @sliding-complete?))
-                                 (reset! gestures-disabled? true)
-                                 (complete-animation sliding-complete?)))))
-        (gesture/on-end (fn [event]
-                          (let [x-translation (oops/oget event "translationX")
-                                reached-end?  (>= x-translation track-width)]
-                            (when (not reached-end?)
-                              (reset-track-position x-pos))))))))

--- a/src/quo/components/buttons/slide_button/view.cljs
+++ b/src/quo/components/buttons/slide_button/view.cljs
@@ -11,94 +11,35 @@
     [quo.theme :as quo.theme]
     [react-native.core :as rn]
     [react-native.gesture :as gesture]
-    [react-native.reanimated :as reanimated]
-    [reagent.core :as reagent]))
+    [react-native.reanimated :as reanimated]))
 
-(defn- f-slider
-  [{:keys [disabled?]}]
-  (let [track-width        (reagent/atom nil)
-        sliding-complete?  (reagent/atom false)
-        gestures-disabled? (reagent/atom disabled?)
-        on-track-layout    (fn [evt]
-                             (let [width (oops/oget evt "nativeEvent.layout.width")]
-                               (reset! track-width width)))]
-    (fn [{:keys [on-reset
-                 on-complete
-                 track-text
-                 track-icon
-                 disabled?
-                 customization-color
-                 size
-                 container-style
-                 theme
-                 type
-                 blur?]}]
-      (let [x-pos             (reanimated/use-shared-value 0)
-            dimensions        (partial utils/get-dimensions
-                                       (or @track-width constants/default-width)
-                                       size)
-            interpolate-track (partial animations/interpolate-track
-                                       x-pos
-                                       (dimensions :usable-track)
-                                       (dimensions :thumb))
-            custom-color      (if (= type :danger) :danger customization-color)]
-        (rn/use-effect (fn []
-                         (when @sliding-complete?
-                           (on-complete)))
-                       [@sliding-complete?])
-        (rn/use-effect (fn []
-                         (when on-reset
-                           (reset! sliding-complete? false)
-                           (reset! gestures-disabled? false)
-                           (animations/reset-track-position x-pos)
-                           (on-reset)))
-                       [on-reset])
-        [gesture/gesture-detector
-         {:gesture (animations/drag-gesture x-pos
-                                            gestures-disabled?
-                                            disabled?
-                                            (dimensions :usable-track)
-                                            sliding-complete?)}
-         [reanimated/view
-          {:test-ID   :slide-button-track
-           :style     (merge (style/track {:disabled?           disabled?
-                                           :customization-color custom-color
-                                           :height              (dimensions :track-height)
-                                           :blur?               blur?})
-                             container-style)
-           :on-layout (when-not (some? @track-width)
-                        on-track-layout)}
-          [reanimated/view {:style (style/track-cover interpolate-track)}
-           [rn/view {:style (style/track-cover-text-container @track-width)}
-            [icon/icon track-icon
-             {:color (utils/text-color custom-color theme blur?)
-              :size  20}]
-            [rn/view {:width 4}]
-            [text/text
-             {:weight :medium
-              :size   :paragraph-1
-              :style  (style/track-text custom-color theme blur?)}
-             track-text]]]
-          [reanimated/view
-           {:style (style/thumb-container {:interpolate-track   interpolate-track
-                                           :thumb-size          (dimensions :thumb)
-                                           :customization-color custom-color
-                                           :theme               theme
-                                           :blur?               blur?})}
-           [reanimated/view {:style (style/arrow-icon-container interpolate-track)}
-            [icon/icon :arrow-right
-             {:color colors/white
-              :size  20}]]
-           [reanimated/view
-            {:style (style/action-icon interpolate-track
-                                       (dimensions :thumb))}
-            [icon/icon track-icon
-             {:color colors/white
-              :size  20}]]]]]))))
+(defn drag-gesture
+  [x-pos gestures-disabled? set-gestures-disabled disabled? track-width sliding-complete?
+   set-sliding-complete
+   on-complete reset-fn]
+  (let [gestures-enabled? (not (or disabled? gestures-disabled?))]
+    (-> (gesture/gesture-pan)
+        (gesture/with-test-ID :slide-button-gestures)
+        (gesture/enabled gestures-enabled?)
+        (gesture/min-distance 0)
+        (gesture/on-update (fn [event]
+                             (let [x-translation (oops/oget event "translationX")
+                                   clamped-x     (utils/clamp-value x-translation 0 track-width)
+                                   reached-end?  (>= clamped-x track-width)]
+                               (reanimated/set-shared-value x-pos clamped-x)
+                               (when (and reached-end? (not sliding-complete?))
+                                 (set-gestures-disabled true)
+                                 (set-sliding-complete true)
+                                 (when on-complete (on-complete reset-fn))))))
+        (gesture/on-end (fn [event]
+                          (let [x-translation (oops/oget event "translationX")
+                                reached-end?  (>= x-translation track-width)]
+                            (when (not reached-end?)
+                              (animations/reset-track-position x-pos))))))))
 
-(defn- view-internal
+(defn view
   "Options
-  - `on-complete`         Callback called when the sliding is complete
+  - `on-complete`         Callback called when the sliding is complete, returns reset fn as a parameter
   - `disabled?`           Boolean that disables the button
                           (_and gestures_)
   - `size`                :size/s-40`/`:size/s-48`
@@ -106,9 +47,78 @@
   - `track-icon`          Key of the icon shown on the track
                           (e.g. `:face-id`)
   - `customization-color` Customization color
-  - `on-reset`            A callback which can be used to reset the component and run required functionality
   "
-  [props]
-  [:f> f-slider props])
-
-(def view (quo.theme/with-theme view-internal))
+  [{:keys [on-complete track-text track-icon disabled? customization-color size
+           container-style type blur?]}]
+  (let [theme                         (quo.theme/use-theme-value)
+        x-pos                         (reanimated/use-shared-value 0)
+        [track-width set-track-width] (rn/use-state nil)
+        [sliding-complete?
+         set-sliding-complete]        (rn/use-state false)
+        [gestures-disabled?
+         set-gestures-disabled]       (rn/use-state disabled?)
+        on-track-layout               (rn/use-callback
+                                       #(set-track-width (oops/oget % "nativeEvent.layout.width")))
+        reset-fn                      (rn/use-callback
+                                       (fn []
+                                         (set-sliding-complete false)
+                                         (set-gestures-disabled false)
+                                         (animations/reset-track-position x-pos)))
+        dimensions                    (rn/use-callback
+                                       (partial utils/get-dimensions
+                                                (or track-width constants/default-width)
+                                                size)
+                                       [track-width])
+        interpolate-track             (rn/use-callback
+                                       (partial animations/interpolate-track
+                                                x-pos
+                                                (dimensions :usable-track)
+                                                (dimensions :thumb))
+                                       [dimensions])
+        custom-color                  (if (= type :danger) :danger customization-color)
+        gesture                       (rn/use-memo #(drag-gesture x-pos
+                                                                  gestures-disabled?
+                                                                  set-gestures-disabled
+                                                                  disabled?
+                                                                  (dimensions :usable-track)
+                                                                  sliding-complete?
+                                                                  set-sliding-complete
+                                                                  on-complete
+                                                                  reset-fn)
+                                                   [gestures-disabled? sliding-complete? disabled?])]
+    [gesture/gesture-detector
+     {:gesture gesture}
+     [reanimated/view
+      {:test-ID   :slide-button-track
+       :style     (merge (style/track {:disabled?           disabled?
+                                       :customization-color custom-color
+                                       :height              (dimensions :track-height)
+                                       :blur?               blur?})
+                         container-style)
+       :on-layout on-track-layout}
+      [reanimated/view {:style (style/track-cover interpolate-track)}
+       [rn/view {:style (style/track-cover-text-container track-width)}
+        [icon/icon track-icon
+         {:color (utils/text-color custom-color theme blur?)
+          :size  20}]
+        [rn/view {:width 4}]
+        [text/text
+         {:weight :medium
+          :size   :paragraph-1
+          :style  (style/track-text custom-color theme blur?)}
+         track-text]]]
+      [reanimated/view
+       {:style (style/thumb-container {:interpolate-track   interpolate-track
+                                       :thumb-size          (dimensions :thumb)
+                                       :customization-color custom-color
+                                       :theme               theme
+                                       :blur?               blur?})}
+       [reanimated/view {:style (style/arrow-icon-container interpolate-track)}
+        [icon/icon :arrow-right
+         {:color colors/white
+          :size  20}]]
+       [reanimated/view
+        {:style (style/action-icon interpolate-track (dimensions :thumb))}
+        [icon/icon track-icon
+         {:color colors/white
+          :size  20}]]]]]))

--- a/src/quo/components/buttons/wallet_button/view.cljs
+++ b/src/quo/components/buttons/wallet_button/view.cljs
@@ -4,26 +4,24 @@
     [quo.components.icon :as quo.icons]
     [quo.foundations.colors :as colors]
     [quo.theme :as theme]
-    [react-native.core :as rn]
-    [reagent.core :as reagent]))
+    [react-native.core :as rn]))
 
-(defn- view-internal
-  []
-  (let [pressed? (reagent/atom false)]
-    (fn
-      [{:keys [on-press on-long-press disabled? icon accessibility-label container-style theme]}]
-      [rn/pressable
-       {:accessibility-label (or accessibility-label :wallet-button)
-        :on-press            on-press
-        :on-press-in         #(reset! pressed? true)
-        :on-press-out        #(reset! pressed? nil)
-        :on-long-press       on-long-press
-        :disabled            disabled?
-        :style               (merge (style/main {:pressed?  @pressed?
-                                                 :theme     theme
-                                                 :disabled? disabled?})
-                                    container-style)}
-       [quo.icons/icon icon
-        {:color (colors/theme-colors colors/neutral-100 colors/white theme)}]])))
-
-(def view (theme/with-theme view-internal))
+(defn view
+  [{:keys [on-press on-long-press disabled? icon accessibility-label container-style]}]
+  (let [theme                  (theme/use-theme-value)
+        [pressed? set-pressed] (rn/use-state false)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:accessibility-label (or accessibility-label :wallet-button)
+      :on-press            on-press
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out
+      :on-long-press       on-long-press
+      :disabled            disabled?
+      :style               (merge (style/main {:pressed?  pressed?
+                                               :theme     theme
+                                               :disabled? disabled?})
+                                  container-style)}
+     [quo.icons/icon icon
+      {:color (colors/theme-colors colors/neutral-100 colors/white theme)}]]))

--- a/src/status_im/common/bottom_sheet/view.cljs
+++ b/src/status_im/common/bottom_sheet/view.cljs
@@ -9,7 +9,6 @@
     [react-native.gesture :as gesture]
     [react-native.hooks :as hooks]
     [react-native.reanimated :as reanimated]
-    [reagent.core :as reagent]
     [status-im.common.bottom-sheet.style :as style]
     [utils.re-frame :as rf]))
 
@@ -57,83 +56,77 @@
            (show translate-y bg-opacity)
            (hide translate-y bg-opacity window-height on-close))))))
 
-(defn- f-view
-  [_ _]
-  (let [sheet-height (reagent/atom 0)
-        item-height  (reagent/atom 0)]
-    (fn [{:keys [hide? insets theme]}
-         {:keys [content selected-item padding-bottom-override border-radius on-close shell?
-                 gradient-cover? customization-color hide-handle? blur-radius]
-          :or   {border-radius 12}}]
-      (let [{window-height :height}           (rn/get-window)
-            bg-opacity                        (reanimated/use-shared-value 0)
-            translate-y                       (reanimated/use-shared-value window-height)
-            sheet-gesture                     (get-sheet-gesture translate-y
-                                                                 bg-opacity
-                                                                 window-height
-                                                                 on-close)
-            selected-item-smaller-than-sheet? (< @item-height
-                                                 (- window-height
-                                                    @sheet-height
-                                                    (:top insets)
-                                                    (:bottom insets)
-                                                    bottom-margin))
-            top                               (- window-height (:top insets) @sheet-height)
-            bottom                            (if selected-item-smaller-than-sheet?
-                                                (+ @sheet-height bottom-margin)
-                                                (:bottom insets))]
-        (rn/use-effect
-         #(if hide?
-            (hide translate-y bg-opacity window-height on-close)
-            (show translate-y bg-opacity))
-         [hide?])
-        (hooks/use-back-handler (fn []
-                                  (when (fn? on-close)
-                                    (on-close))
-                                  (rf/dispatch [:hide-bottom-sheet])
-                                  true))
-        [rn/view {:style {:flex 1}}
-         ;; backdrop
-         [rn/pressable
-          {:on-press #(rf/dispatch [:hide-bottom-sheet])
-           :style    {:flex 1}}
-          [reanimated/view
-           {:style (reanimated/apply-animations-to-style
-                    {:opacity bg-opacity}
-                    {:flex 1 :background-color colors/neutral-100-opa-70})}]]
-         ;; sheet
-         [gesture/gesture-detector {:gesture sheet-gesture}
-          [reanimated/view
-           {:style (reanimated/apply-animations-to-style
-                    {:transform [{:translateY translate-y}]}
-                    (style/sheet insets window-height selected-item))}
-           (when shell?
-             [blur/ios-view
-              {:style         style/shell-bg
-               :blur-radius   (or blur-radius 20)
-               :blur-type     :transparent
-               :overlay-color :transparent}])
-           (when selected-item
-             [rn/view
-              {:on-layout #(reset! item-height (.-nativeEvent.layout.height ^js %))
-               :style
-               (style/selected-item theme top bottom selected-item-smaller-than-sheet? border-radius)}
-              [selected-item]])
+(defn view
+  [{:keys [hide? insets]}
+   {:keys [content selected-item padding-bottom-override border-radius on-close shell?
+           gradient-cover? customization-color hide-handle? blur-radius]
+    :or   {border-radius 12}}]
+  (let [theme                             (quo.theme/use-theme-value)
+        sheet-height                      (rn/use-ref-atom 0)
+        item-height                       (rn/use-ref-atom 0)
+        {window-height :height}           (rn/get-window)
+        bg-opacity                        (reanimated/use-shared-value 0)
+        translate-y                       (reanimated/use-shared-value window-height)
+        sheet-gesture                     (get-sheet-gesture translate-y
+                                                             bg-opacity
+                                                             window-height
+                                                             on-close)
+        selected-item-smaller-than-sheet? (< @item-height
+                                             (- window-height
+                                                @sheet-height
+                                                (:top insets)
+                                                (:bottom insets)
+                                                bottom-margin))
+        top                               (- window-height (:top insets) @sheet-height)
+        bottom                            (if selected-item-smaller-than-sheet?
+                                            (+ @sheet-height bottom-margin)
+                                            (:bottom insets))]
+    (rn/use-effect
+     #(if hide?
+        (hide translate-y bg-opacity window-height on-close)
+        (show translate-y bg-opacity))
+     [hide?])
+    (hooks/use-back-handler (fn []
+                              (when (fn? on-close)
+                                (on-close))
+                              (rf/dispatch [:hide-bottom-sheet])
+                              true))
+    [rn/view {:style {:flex 1}}
+     ;; backdrop
+     [rn/pressable
+      {:on-press #(rf/dispatch [:hide-bottom-sheet])
+       :style    {:flex 1}}
+      [reanimated/view
+       {:style (reanimated/apply-animations-to-style
+                {:opacity bg-opacity}
+                {:flex 1 :background-color colors/neutral-100-opa-70})}]]
+     ;; sheet
+     [gesture/gesture-detector {:gesture sheet-gesture}
+      [reanimated/view
+       {:style (reanimated/apply-animations-to-style
+                {:transform [{:translateY translate-y}]}
+                (style/sheet insets window-height selected-item))}
+       (when shell?
+         [blur/ios-view
+          {:style         style/shell-bg
+           :blur-radius   (or blur-radius 20)
+           :blur-type     :transparent
+           :overlay-color :transparent}])
+       (when selected-item
+         [rn/view
+          {:on-layout #(reset! item-height (.-nativeEvent.layout.height ^js %))
+           :style
+           (style/selected-item theme top bottom selected-item-smaller-than-sheet? border-radius)}
+          [selected-item]])
 
-           [rn/view
-            {:style     (style/sheet-content theme padding-bottom-override insets shell? bottom-margin)
-             :on-layout #(reset! sheet-height (.-nativeEvent.layout.height ^js %))}
-            (when (and gradient-cover? customization-color)
-              [rn/view {:style style/gradient-bg}
-               [quo/gradient-cover
-                {:customization-color customization-color
-                 :opacity             0.4}]])
-            (when-not hide-handle?
-              [quo/drawer-bar])
-            [content]]]]]))))
-
-(defn- internal-view
-  [args sheet]
-  [:f> f-view args sheet])
-
-(def view (quo.theme/with-theme internal-view))
+       [rn/view
+        {:style     (style/sheet-content theme padding-bottom-override insets shell? bottom-margin)
+         :on-layout #(reset! sheet-height (.-nativeEvent.layout.height ^js %))}
+        (when (and gradient-cover? customization-color)
+          [rn/view {:style style/gradient-bg}
+           [quo/gradient-cover
+            {:customization-color customization-color
+             :opacity             0.4}]])
+        (when-not hide-handle?
+          [quo/drawer-bar])
+        [content]]]]]))

--- a/src/status_im/common/standard_authentication/standard_auth/slide_button/view.cljs
+++ b/src/status_im/common/standard_authentication/standard_auth/slide_button/view.cljs
@@ -3,45 +3,32 @@
     [quo.core :as quo]
     [quo.theme :as quo.theme]
     [react-native.core :as rn]
-    [reagent.core :as reagent]
     [status-im.common.standard-authentication.standard-auth.authorize :as authorize]
     [status-im.constants :as constants]
     [utils.re-frame :as rf]))
 
-(defn- view-internal
-  [_]
-  (let [reset-slider?   (reagent/atom false)
-        on-close        (fn []
-                          (js/setTimeout
-                           #(reset! reset-slider? true)
-                           200))
+(defn view
+  [{:keys [track-text customization-color auth-button-label on-auth-success on-auth-fail
+           auth-button-icon-left size blur? container-style]
+    :or   {container-style {:flex 1}}}]
+  (let [theme           (quo.theme/use-theme-value)
         auth-method     (rf/sub [:auth-method])
-        biometric-auth? (= auth-method constants/auth-method-biometric)]
-    (fn [{:keys [track-text
-                 customization-color
-                 auth-button-label
-                 on-auth-success
-                 on-auth-fail
-                 auth-button-icon-left
-                 size
-                 theme
-                 blur?
-                 container-style]
-          :or   {container-style {:flex 1}}}]
-      [rn/view {:style container-style}
-       [quo/slide-button
-        {:size                size
-         :customization-color customization-color
-         :on-reset            (when @reset-slider? #(reset! reset-slider? false))
-         :on-complete         #(authorize/authorize {:on-close              on-close
-                                                     :auth-button-icon-left auth-button-icon-left
-                                                     :theme                 theme
-                                                     :blur?                 blur?
-                                                     :biometric-auth?       biometric-auth?
-                                                     :on-auth-success       on-auth-success
-                                                     :on-auth-fail          on-auth-fail
-                                                     :auth-button-label     auth-button-label})
-         :track-icon          (if biometric-auth? :i/face-id :password)
-         :track-text          track-text}]])))
-
-(def view (quo.theme/with-theme view-internal))
+        biometric-auth? (= auth-method constants/auth-method-biometric)
+        on-complete     (rn/use-callback
+                         (fn [reset]
+                           (authorize/authorize {:on-close              #(js/setTimeout reset 200)
+                                                 :auth-button-icon-left auth-button-icon-left
+                                                 :theme                 theme
+                                                 :blur?                 blur?
+                                                 :biometric-auth?       biometric-auth?
+                                                 :on-auth-success       on-auth-success
+                                                 :on-auth-fail          on-auth-fail
+                                                 :auth-button-label     auth-button-label}))
+                         [theme])]
+    [quo/slide-button
+     {:container-style     container-style
+      :size                size
+      :customization-color customization-color
+      :on-complete         on-complete
+      :track-icon          (if biometric-auth? :i/face-id :password)
+      :track-text          track-text}]))

--- a/src/status_im/contexts/preview/quo/buttons/slide_button.cljs
+++ b/src/status_im/contexts/preview/quo/buttons/slide_button.cljs
@@ -48,7 +48,7 @@
            :disabled?           (:disabled? @state)
            :blur?               @blur?
            :type                (:type @state)
-           :on-complete         (fn []
+           :on-complete         (fn [_]
                                   (js/setTimeout (fn [] (reset! complete? true))
                                                  1000)
                                   (js/alert "I don't wanna slide anymore"))}]

